measurement/plugins/netflix_fast/measurements.py

```diff
@@ -0,0 +1,334 @@
+"""
+Using the netflix_fast v2 api, the test collects some details about the client, and launches 1 thread per provided URL to download. Every `sleep_seconds` (presently 0.2) the test will append the latest speed (calculated by total downloaded bytes/total time taken) before checking for, in order:
+    - `max_time` (presently 30s) expired.
+    - Results have become "stabilised"
+    - All threads have finished downloading
+    - A single thread has finished downloading, IF `terminate_on_thread_complete=True`
+
+Stabilisation is considered to be:
+    - Downloaded has been running longer than `MIN_TIME` (presently 3s)
+    - AND more than or equal to `STABLE_MEASUREMENTS_LENGTH` (presently 6) have been recorded
+    - AND maximum percentage delta in these measurements is `< STABLE_MEASUREMENTS_DELTA` presently (2%)
+
+In cases where the test concludes independently of the main loop (i.e when `reason_terminated == "thread_complete"`) The speed at the instant the thread completes is used, otherwise the final speed is used.
+
+All this is then packaged into a `NetflixFastMeasurementResult`
+
+After this, each of the URLs downloaded from has a latency test then have an Honesty-Box LatencyMeasurement test run against them, the results of which, along with the location and download rates/sizes for each thread are put into a `NetflixFastThreadResult`.
+
+All these results are then returned as a list.
+"""
+
+import requests
+import re
+import time
+import urllib
+import json
+from threading import Thread
+from collections import deque
+from statistics import mean
+
+from measurement.measurements import BaseMeasurement
+from measurement.results import Error
+from measurement.units import RatioUnit, TimeUnit, StorageUnit, NetworkUnit
+from measurement.plugins.latency.measurements import LatencyMeasurement
+from measurement.plugins.download_speed.measurements import DownloadSpeedMeasurement
+from measurement.plugins.netflix_fast.results import (
+    NetflixFastMeasurementResult,
+    NetflixFastThreadResult,
+)
+
+
+NETFLIX_ERRORS = {
+    "netflix-err": "Netflix test encountered an unknown error",
+    "netflix-ping": "Netflix test encountered an error when pinging hosts",
+    "netflix-response": "Netflix test received an invalid response from fast.com",
+    "netflix-script-regex": "Netflix test failed to find script in the response",
+    "netflix-script-response": "Netflix test received an invalid response from script",
+    "netflix-token-regex": "Netflix test failed to find token in the response",
+    "netflix-api-response": "Netflix test received an invalid response when querying for URLs",
+    "netflix-api-json": "Netflix test failed to decode URLs",
+    "netflix-api-parse": "Netflix test failed interpret elements of the decoded JSON",
+    "netflix-connection": "Netflix test failed to connect to download URLs",
+    "netflix-download": "Netflix test encountered an error downloading data",
+}
+CHUNK_SIZE = 64 * 2 ** 10
+MIN_TIME = 3
```

Comment:
Is there a unit associated with `MIN_TIME`? I am not sure what this variable is from readying the name